src/firefly/js/ui/dynamic/DLGeneratedDropDown.js

```diff
@@ -489,20 +494,33 @@ function DLGeneratedTableSearch({currentTblId, initArgs, sideBar, regHasUrl, url
 
 
     const [,setCallId]= useState('none');
-    const {current:clickFuncRef} = useRef({clickFunc:undefined});
+    const [{onClick},setClickFuncImpl]= useState({});
     const qAna= analyzeQueries(currentTblId);
     const tabsKey= 'Tabs-'+currentTblId;
     const matchUrl= initArgs?.urlApi?.url?.[0]===url;
 
+    const setClickFunc= (obj) => {
+        if (obj?.onClick && !onClick) setClickFuncImpl(obj);
+    };
+
+    if (initArgs.urlApi?.url) {
+        console.log(`DLGeneratedTableSearch: (via api): regLoaded:${regLoaded}, regHasUrl:${regHasUrl}, matchUrl:${matchUrl}, clickFunc:${Boolean(onClick)}, ${url}`,
+            initArgs.urlApi);
+    }
+
+
     useEffect(() => {
-        if (initArgs?.urlApi?.execute && clickFuncRef.clickFunc && matchUrl) {
+        console.log(`in effect: regLoaded:${regLoaded}, regHasUrl:${regHasUrl}, matchUrl:${matchUrl}, clickFunc:${Boolean(onClick)}`);
+        if (initArgs?.urlApi?.execute && onClick && matchUrl) {
+            console.log('trying: execute once');
             executeInitOnce(true, () => {
+                console.log(`execute (${initArgs.urlApi.callId}) url: ${initArgs?.urlApi.url}`, initArgs?.urlApi);
                 setCallId(initArgs.urlApi.callId ?? 'none'); //forces one more render after unmount
                 dispatchMountFieldGroup(GROUP_KEY, false, false); // unmount to force to forget default so it will reinit
-                setTimeout(() => clickFuncRef?.clickFunc?.(),10);
+                setTimeout(() => onClick?.(),10);
```

Comment:
we don't have to do chaining `onClick?` here as we already tested it in if-condition above

Reply:
thanks. fixed.